### PR TITLE
Add `closed_boundary` argument to `hamiltonians.models.TFIM`

### LINKED ISSUE #433: Ising Hamiltonian with non-periodic boundary conditions
Currently, the Transverse Field Ising Hamiltonian (`qibo.hamiltonians.TFIM`) only supports periodic boundary conditions in its definition. Would it be possible to allow for non-periodic boundary conditions as well, e.g. by passing a boolean argument `periodic=False`?

### PR DESCRIPTION
**Pull request by renatomello**
Closes #433 

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

File: src/qibo/hamiltonians/models.py [hunk 1]
```diff
@@ -46,29 +51,39 @@ def Y(nqubits, dense: bool = True, backend=None):
...
         H = - \\sum _{k=0}^{N} \\, Z_{k} \\, .
 
     Args:
         nqubits (int): number of qubits.
-        dense (bool): If ``True`` it creates the Hamiltonian as a
+        dense (bool, optional): If ``True`` it creates the Hamiltonian as a
             :class:`qibo.core.hamiltonians.Hamiltonian`, otherwise it creates
             a :class:`qibo.core.hamiltonians.SymbolicHamiltonian`.
-        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
+       backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
```
Comment (renatomello) [inline src/qibo/hamiltonians/models.py]:
```suggestion
        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
```
